src/vm/arm64/stubs.cpp

```diff
@@ -1995,8 +2010,112 @@ PCODE DynamicHelpers::CreateDictionaryLookupHelper(LoaderAllocator * pAllocator,
 {
     STANDARD_VM_CONTRACT;
 
-    // TODO (NYI)
-    ThrowHR(E_NOTIMPL);
+    // It's available only via the run-time helper function
+    if (pLookup->indirections == CORINFO_USEHELPER)
+    {
+        BEGIN_DYNAMIC_HELPER_EMIT(32);
+
+        // X0 already contains generic context parameter
+        // reuse EmitHelperWithArg for below two operations
+        // X1 <- pLookup->signature
+        // branch to pLookup->helper
+        EmitHelperWithArg(p, pAllocator, (TADDR)pLookup->signature, CEEJitInfo::getHelperFtnStatic(pLookup->helper));
+
+        END_DYNAMIC_HELPER_EMIT();
+    }
+    else
+    {
+        int indirectionsCodeSize = 0;
+        int indirectionsDataSize = 0;
+        for (WORD i = 0; i < pLookup->indirections; i++) {
+            indirectionsCodeSize += (pLookup->offsets[i] > 32760 ? 8 : 4); // if( > 32760) (8 code bytes) else 4 bytes for instruction with offset encoded in instruction
+            indirectionsDataSize += (pLookup->offsets[i] > 32760 ? 4 : 0);
+        }
+
+        int codeSize = indirectionsCodeSize + (pLookup->testForNull ? 44 : 4);
+        if(pLookup->testForNull)
+        {
+            //padding for 8-byte align (required by EmitHelperWithArg)
+            if((codeSize & 0x7) != 0)
```

Comment:
Ok. I got it. I interpret this way.  If indirectionsCodeSize (input to EmithelperWithArg) is not aligned, we don’t pad since the size of EmitHelpersWithArg is not aligned resulting in total code size is aligned. I think we should add comments about the numbers how they come from at minimum.
```
    int codeSize = indirectionsCodeSize;
    if (pLookup->testForNull)
    {
        int padding = (indirectionsCodeSize & 0x7) == 0 ? 4 : 0;
        codeSize += 4 /* mov */ + 12 /* cbz-ret-mov */+ 28 /* size of EmitHelperWithArg */ + padding;
    }
    else
    {
        codeSize += 4 /* ret */;
    }
```